p2p/src/net/libp2p/behaviour.rs

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): A. Altonen
+
+//! Network behaviour configuration for libp2p
+
+use crate::{
+    message,
+    net::{
+        self,
+        libp2p::{
+            constants::*,
+            sync::*,
+            types::{self, ConnectivityEvent, Libp2pBehaviourEvent, PubSubEvent},
+        },
+        types::PubSubTopic,
+    },
+};
+use common::chain::config::ChainConfig;
+use libp2p::{
+    core::PeerId,
+    gossipsub::{self, Gossipsub, GossipsubConfigBuilder, MessageAuthenticity, ValidationMode},
+    identify, identity, mdns, ping,
+    request_response::*,
+    swarm::{
+        ConnectionHandler, IntoConnectionHandler, NetworkBehaviour as Libp2pNetworkBehaviour,
+        NetworkBehaviourAction, NetworkBehaviourEventProcess, PollParameters,
+    },
+    NetworkBehaviour,
+};
+use logging::log;
+use serialization::Decode;
+use std::{
+    collections::{HashMap, HashSet, VecDeque},
+    iter,
+    num::NonZeroU32,
+    sync::Arc,
+    task::{Context, Poll, Waker},
+};
+
+#[derive(NetworkBehaviour)]
+#[behaviour(
+    out_event = "Libp2pBehaviourEvent",
+    event_process = true,
+    poll_method = "poll"
+)]
+pub struct Libp2pBehaviour {
```

Comment:
Could you please document this a little bit, explaining what "behavior" really means with some examples?

Reply:
This seems to be a part of how libp2p works. So, never mind.